### PR TITLE
vm: Add dusk-vm

### LINKED ISSUE #3235: vm: Add a `dusk-vm` workspace member
#### Summary
Add a `dusk-vm` workspace member that will wrap around `piecrust` vm, enriching it with features specifically needed by the dusk-network.

#### Detailed Description
The rusk-abi workspace member will be split into two parts:
- [x] the abi part will move to `dusk-core` under an `abi` feature-flag #3198
- [ ] the remaining host part in rusk-abi will be renamed to `dusk-vm` and a specific Dusk `VM` will be added that enriches the piecrust-VM with the host-functions specified by the dusk-ABI #3237

### PR DESCRIPTION
**Pull request by moCello**
To help reviewing these changes, I've split up the work into this one and:

- [x] #3239

Resolves #3235

### CODE REVIEW

**Review by HDauven — APPROVED** [review]

**Review by miloszm — APPROVED** [review]

File: vm/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+pub fn new_genesis_session(vm: &VM, chain_id: u8) -> Session {
+    vm.session(
+        SessionData::builder()
+            .insert(Metadata::CHAIN_ID, chain_id)
+            .expect("Inserting chain ID in metadata should succeed")
+            .insert(Metadata::BLOCK_HEIGHT, 0)
+            .expect("Inserting block height in metadata should succeed"),
+    )
+    .expect("Creating a genesis session should always succeed")
+}
```
Comment (herr-seppia) [inline vm/src/lib.rs:62]: Just a reminder for future refactors

These should be part of `Impl VM` and take `&self`

File: vm/src/cache.rs
Comment (herr-seppia) [inline vm/src/cache.rs:1]: This file use `RUSK_ABI_` prefix for cache size environment variable

File: vm/rustfmt.toml
Comment (herr-seppia) [inline vm/rustfmt.toml:1]: this file can be removed

**Review by herr-seppia — APPROVED** [review]
LGTM

**Comment by moCello** [comment]
@herr-seppia good finds! I've created #3273 to pick up your comments at a later point!

**Pull request merged**